Find a functionally equivalent code:android.graphics.Paint_Delegate.updateFontObject:COMMENT
Method Modifier: private     
Comment:/**
 * Update the {@link Font} object from the typeface, text size and scaling
 */

@@ -1,11 +1,17 @@
 {
     if (mTypeface != null) {
         // Get the fonts from the TypeFace object.
-        List<Font> fonts = mTypeface.getFonts();
+        List<Font> fonts = mTypeface.getFonts(mFontVariant);
         // create new font objects as well as FontMetrics, based on the current text size
         // and skew info.
         ArrayList<FontInfo> infoList = new ArrayList<FontInfo>(fonts.size());
         for (Font font : fonts) {
+            if (font == null) {
+                // If the font is null, add null to infoList. When rendering the text, if this
+                // null is reached, a warning will be logged.
+                infoList.add(null);
+                continue;
+            }
             FontInfo info = new FontInfo();
             info.mFont = font.deriveFont(mTextSize);
             if (mTextScaleX != 1.0 || mTextSkewX != 0) {

